site/src/pages/search.js

```diff
@@ -22,47 +24,55 @@ export default class Search extends React.Component {
     let paginationObject = new PaginationObject()
     this.state = {
       query: ``,
+      sanitizedQuery:``,
       results: [],
-      paginationObject: paginationObject
+      paginationObject: paginationObject,
     }
     this.data = props.data
-    this.trackSiteSearch = debounce(searchAnalytics.trackSiteSearch, props.debounceDelay)
+    this.trackSiteSearch = debounce(
+      searchAnalytics.trackSiteSearch,
+      props.debounceDelay
+    )
     this.updateSearchResults = this.updateSearchResults.bind(this)
     this.updatePagination = this.updatePagination.bind(this)
+
+    this.querySanitizer = new QuerySanitizer(dictionaryDefinition)
   }
 
   updatePagination(pageTarget) {
     this.state.paginationObject.goToPage(pageTarget)
     // update state to get page to rerender
     this.setState({
-      paginationObject: this.state.paginationObject
+      paginationObject: this.state.paginationObject,
     })
   }
 
   updateSearchResults(evt) {
     this.state.paginationObject.goToPage(0)
 
-    const query = evt.target.value
+    const query = this.querySanitizer.sanitize(evt.target.value)
```

Comment:
Would prefer a "const rawQuery" and another "const sanitizedQuery" to explain the connection here, and below in the setState() call.

Reply:
Done